airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/abstract_stream_state_converter.py

```diff
@@ -121,10 +121,13 @@ def merge_intervals(self, intervals: List[MutableMapping[str, Any]]) -> List[Mut
             return []
 
         sorted_intervals = sorted(intervals, key=lambda x: (x[self.START_KEY], x[self.END_KEY]))
+        sorted_intervals = [interval for interval in sorted_intervals if interval.get("most_recent_record") is not None]
```

Comment:
we'll need to make sure to properly handle the case where none of the partitions have a most recent record. 

Unfortunately, the end result is we can't update the state.